### PR TITLE
PostgreSQL, web DB heroku connection and auto migrations

### PR DESCRIPTION
**Pull request by Archanial**
Adds support for postgresql, dynamic connection string generation for heroku and auto-migrations on startup (moved from seed).

Works on live, but do not merge before I delete the database on heroku.

New project has been created to hold postgre migrations.

Some important stuff in `OnModelCreating` has been changed, most notably:
- you can't use "datetime" anymore
- you can't use nvarchar(max) (or varchar(max))
- every text field should not use unicode (not supported by postgre)
Old MySql migrations have been deleted and new one has been created.

### CODE REVIEW

File: src/Patronage.Api/Program.cs [hunk 1]
```diff
@@ -52,6 +94,42 @@
 
     var app = builder.Build();
 
+    ApplyMigrations();
+
+    void ApplyMigrations()
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Program.cs:99]: Move method to extension method. There is a mess in code and we don't know what is a method and what is the main code  method.

File: src/Patronage.Api/Program.cs [hunk 2]
```diff
@@ -52,6 +94,42 @@
...
+                logger.Info($"You are on schema version: {lastAppliedMigration}");
+            }
+            catch (Exception ex)
+            {
+                logger.Error(ex);
+            }
+        }
+    }
+
     SendData(app);
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Program.cs:133]: Why we try to run migrations multiple times?

File: src/Patronage.Api/Program.cs [hunk 3]
```diff
@@ -86,13 +164,13 @@ void SendData(IHost app)
 
     app.MapControllers();
 
-    logger.Debug("Initializing complete!");
+    logger.Info("Initializing complete!");
     string? port = Environment.GetEnvironmentVariable("PORT");
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Program.cs:168]: Use ?? operator instead of if statement.

File: src/Patronage.Api/appsettings.json [hunk 4]
```diff
@@ -14,6 +14,7 @@
   "AllowedHosts": "*",
   "ConnectionStrings": {
     "Default": "Data Source=(localdb)\\MSSQLLocalDB;Initial Catalog=patronageDB;Integrated Security=True;",
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/appsettings.json:16]: We should have one connection string if only one database is used. Perfectly we should get connection string from Environment Variable and if not exists get default connection string.

File: src/Patronage.Api/Program.cs [hunk 5]
```diff
@@ -26,13 +28,53 @@
         c.SwaggerDoc("v1", new OpenApiInfo { Title = "Patronage 2022 API", Version = "v1" });
     });
 
-    builder.Services.AddDbContext<TableContext>((DbContextOptionsBuilder options) =>
+    ///This determines which connection string are we going to use
+    ///If environmental variable "DATABASE_URL" is set we will build connection string to connect to remove database
+    ///Won't work on local! Else uses our "Default connection string.
+    ///Why do we have to build connection string dynamically? Heroku periodically changes credentials, so we have to keep up with that.
+    if(Environment.GetEnvironmentVariable("DATABASE_URL") != null || builder.Configuration.GetValue("provider", "mysql").ToLower() == "postgre")
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Program.cs:35]: String comparision should be done using Equals method and use InvariantCultureIgnoringCase.

File: src/Patronage.Api/Program.cs [hunk 6]
```diff
@@ -26,13 +28,53 @@
         c.SwaggerDoc("v1", new OpenApiInfo { Title = "Patronage 2022 API", Version = "v1" });
     });
 
-    builder.Services.AddDbContext<TableContext>((DbContextOptionsBuilder options) =>
+    ///This determines which connection string are we going to use
+    ///If environmental variable "DATABASE_URL" is set we will build connection string to connect to remove database
+    ///Won't work on local! Else uses our "Default connection string.
```
Comment (mateuszgiedzinski) [inline src/Patronage.Api/Program.cs:33]: I recommend to move database configuration code to other class maybe app extension.

**Review by mateuszgiedzinski — DISMISSED** [review]
I recommend some changes for reading clarity. Also try to squash your commit. to remove this strange commits I mention  on slack.

File: src/Patronage.Api/appsettings.json (hunk 4 above)
Comment (Archanial) [reply src/Patronage.Api/appsettings.json:16]: It's not possible, postgre uses different format of connection string

File: src/Patronage.Api/Program.cs [hunk 7]
```diff
@@ -86,13 +164,13 @@ void SendData(IHost app)
 
     app.MapControllers();
 
-    logger.Debug("Initializing complete!");
+    logger.Info("Initializing complete!");
     string? port = Environment.GetEnvironmentVariable("PORT");
     if(port == null)
     {
         port = "80";
     }
```
Comment (Archanial) [inline src/Patronage.Api/Program.cs:172]:
```suggestion
    string? port = Environment.GetEnvironmentVariable("PORT") ?? "80";
```

File: src/Patronage.Api/Program.cs (hunk 2 above)
Comment (Archanial) [reply src/Patronage.Api/Program.cs:133]: We're not, I've removed migrate part from `Seed()` method.
Program.cs runs only once during app startup.

File: src/Patronage.Api/Program.cs [hunk 8]
```diff
@@ -10,11 +10,13 @@
...
 using Patronage.Api.Middleware;
+using Npgsql;
+
 
 try
 {
-    var logger = NLog.LogManager.Setup().LoadConfigurationFromAppSettings().GetCurrentClassLogger();
-    logger.Debug("init main");
+    var logger = LogManager.Setup().LoadConfigurationFromAppSettings().GetCurrentClassLogger();
+    logger.Info("Starting initializing");
```
Comment (Archanial) [inline src/Patronage.Api/Program.cs:19]:
```suggestion
    logger.Info("Starting");
```

**Pull request merged**